generator/mapper.go

```diff
@@ -74,62 +75,84 @@ func CreateCatalogsFromProfile(profileArg *profile.Profile) ([]*catalog.Catalog,
 		}
 	}
 }
+func getSubControl(call profile.Call, ctrls []catalog.Control, helper impl.Catalog) (*catalog.Subcontrol, error) {
```

Comment:
@minhaj10p why the need to return a pointer to a subcontrol here?

Reply:
just so I could return a nil if not found. nothing special. should I remove the pointer?

Reply:
Ah gotcha. Yea in this case it's probably fine just to return an empty struct vs. having to dereference the pointer every time in the rest of your code.

Reply:
Changes pushed @anweiss